EpikCord/flags.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List
+
+
+class Flag:
+    class_flags: Dict[str, int]
+
+    def __init_subclass__(cls) -> None:
+        cls.class_flags = {k: v for k, v in cls.__dict__.items() if isinstance(v, int)}
+
+    def __init__(self, value: int = 0, **kwargs):
+        self.value = value
+        self.turned_on: List[str] = [k for k, a in kwargs.items() if a]
+
+        for k, v in self.class_flags.items():
+            if v & value and k not in self.turned_on:
+                self.turned_on.append(k)
+
+        self.calculate_from_turned()
+
+    def calculate_from_turned(self):
+        value = 0
+        for key, flag in self.class_flags.items():
+            if key in self.class_flags:
+                value |= flag
+        self.value = value
+
+    def __getattribute__(self, __name: str) -> Any:
+        original = super().__getattribute__
+        if __name in original("class_flags"):
+            return __name in original("turned_on")
+        return original(__name)
+
+    def __setattr__(self, __name: str, __value: Any) -> None:
+        if __name not in self.class_flags:
+            return super().__setattr__(__name, __value)
+        if __value and __name not in self.turned_on:
+            self.turned_on.append(__name)
+        elif not __value and __name in self.turned_on:
+            self.turned_on.remove(__name)
+        self.calculate_from_turned()
+
+    @classmethod
+    def all(cls):
+        return cls(**{k: True for k in cls.class_flags})
+
+
+class Intents(Flag):
+    guilds = 1 << 0
+    members = 1 << 1
+    bans = 1 << 2
+    emojis_and_stickers = 1 << 3
+    integrations = 1 << 4
+    webhooks = 1 << 5
+    invites = 1 << 6
+    voice_states = 1 << 7
+    presences = 1 << 8
+
+    guild_messages = 1 << 9
+    guild_message_reactions = 1 << 10
+    guild_message_typing = 1 << 11
+
+    direct_messages = 1 << 12
+    direct_message_reactions = 1 << 13
+    direct_message_typing = 1 << 14
+
+    message_content = 1 << 15
+    scheduled_event = 1 << 16
+
+    auto_moderation_configuration = 1 << 20
+    auto_moderation_action_execution = 1 << 21
```

Comment:
names are getting long, i suggest to keep them short, like `automod_conf` & `automod_exec`

Reply:
I don't think that the name length matters; Discord set it as so, so we're just matching the docs. I mean we can consider aliases at a later date.